scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -309,284 +310,453 @@ Note that while we applied this technique to assign cell types using the ADT dat
+        "Th17 cells", 
+        "Th2 cells",
+        "Follicular helper T cells"),
+    "B cells" = c(
+        "Naive B cells",
+        "Switched memory B cells",
+        "Non-switched memory B cells",
+        "Exhausted B cells",
+        "Plasmablasts"		
+    )
+  ) |>
+  # order for plotting
+  forcats::fct_relevel(
+    "Helper T cells",
+    "T regulatory cells",
+    "Naive CD4 T cells",
+    "Terminal effector CD4 T cells",
+    "Naive CD8 T cells",
+    "Central memory CD8 T cells",
+    "Effector memory CD8 T cells",
+    "Terminal effector CD8 T cells",
+    "T cells",
+    "Natural killer cells",
+    "B cells",
+    "Monocytes",
+    "Dendritic cells",
+    "Progenitor cells",
+    "Low-density basophils"
+  )
 ```
 
-Now we plot.
+Now that we have that set up, we can plot using our collapsed and ordered cell type labels.
 
-```{r}
+```{r plot collapsed, live=TRUE}
 sce$celltype_collapsed <- collapsed_labels
-scater::plotUMAP(sce, colour_by = "celltype_collapsed")
+scater::plotUMAP(sce, 
+                 colour_by = "celltype_collapsed")
 ```
 
 
 ### Heatmap of cell types & clusters
 
+Let's look at how the cell type assignments we obtained using `SingleR` compare to the clusters that we found using the unsupervised clustering at the start of this notebook.
 
-Let's look at how the cell type assignments we have compare to the clusters we found from the data alone.
-
-To do this, we will first use `table` function to build a contingency table of the cell types and clusters that each cell was classified with. 
+To do this, we will again use the `table()` function, but now with two vectors as input, to build a contingency table of the cell types and clusters that each cell was classified with. 
 
-```{r}
+```{r type cluster table}
 # create a table of clusters & cell type counts
 type_cluster_tab <- table(sce$celltype_fine, sce$nn_cluster, useNA = "ifany")
-type_cluster_tab
+
+# look at the top corner of the results
+type_cluster_tab[1:5, 1:5]
 ```
 
-Since the number of cells in each cluster will vary, we'd like to convert these counts into proportions. 
-We'll do this by dividing each cluster/cell type count by the total number of cells in that cluster.
+As you can see, this produced a table with rows for each cell type and columns for each cluster number.
+The values are the count of cells for each cluster/cell type combination.
+These raw counts of cells are not quite what we are going to want for visualization. 
+Since the total number of cells in each cluster will vary, we'd like to convert these counts into the proportions of each cell type in each cluster.
 
+We'll do this by going through the table column by column and dividing each value by the sum for that cluster.
+To do that, we will use the `apply` function, which allows us to operate on a matrix row by row or column by column, applying a function to each "slice".
+Since the function we want to apply is very short, we will use R's new anonymous function shorthand: 
```

Comment:
"new anonymous function shorthand" some kind of release link could be good? Mostly I mean, what is "NEW" in this case?

Reply:
How about just this:

```suggestion
Since the function we want to apply is very short, we will use R's new (as of  version 4.1) anonymous function shorthand: 
```